### PR TITLE
Better doc

### PR DESCRIPTION
**Pull request by donald213**

### CODE REVIEW

File: ooni/agent/agent.py [hunk 1]
```diff
@@ -6,9 +6,21 @@
 from ooni.agent.scheduler import SchedulerService
 
 class AgentService(service.MultiService):
+    """Contain any element to manage agent services.
+
+
+    The class above inherits to another class wich is a services container."""
```
Comment (hellais) [inline ooni/agent/agent.py]: I believe this comment is redundant.

File: ooni/agent/agent.py [hunk 2]
```diff
@@ -6,9 +6,21 @@
 from ooni.agent.scheduler import SchedulerService
 
 class AgentService(service.MultiService):
+    """Contain any element to manage agent services.
```
Comment (hellais) [inline ooni/agent/agent.py]: This should read:

"The AgentService is responsible for running the ooniprobe web UI and the scheduled periodic tasks as a system daemon."

File: ooni/agent/agent.py [hunk 3]
```diff
@@ -6,9 +6,21 @@
...
+    The class above inherits to another class wich is a services container."""
+
+
     def __init__(self, web_ui_port):
-        service.MultiService.__init__(self)
+        """Load configuration or made the default configuration of the service.
 
+
+        Set up the child services of the AgentService.
+        If the advanced configuration is not enabled, the page is not displayed.
```
Comment (hellais) [inline ooni/agent/agent.py]: You should specify what is meant by "advanced configuration". You probably want to say:

"If `config.advanced.disabled_webui` is set to True, then the web UI will not be started"

File: ooni/agent/agent.py [hunk 4]
```diff
@@ -22,7 +34,11 @@ def __init__(self, web_ui_port):
 
 
     def startService(self):
+        """Run a service."""
```
Comment (hellais) [inline ooni/agent/agent.py]: This and the comment below are redundant.

File: ooni/agent/scheduler.py [hunk 5]
```diff
@@ -35,6 +35,7 @@ class FileSystemlockAndMutex(object):
     different stacks (threads/fibers) within the same process without races.
     """
     def __init__(self, file_path):
+        """Initialize a lock for the filesystem and a lock for event driven systems."""
```
Comment (hellais) [inline ooni/agent/scheduler.py]: These comments are redundant. If anything I would document what `file_path` is.

File: ooni/ui/web/web.py [hunk 6]
```diff
@@ -6,6 +6,11 @@
 from ooni.settings import config
 
 class WebUIService(service.MultiService):
+    """Contain any element to manage the web user interface service.
+
+
+    The class above inherits to another class wich is a services container."""
```
Comment (hellais) [inline ooni/ui/web/web.py]: Maybe it's sufficient to say. "This multiservice contains the ooniprobe web user interface"

File: ooni/ui/web/web.py [hunk 7]
```diff
@@ -14,16 +19,23 @@ def __init__(self, director, scheduler, port_number=8842):
         self.port_number = port_number
 
     def startService(self):
+        """Start a web user interface.
+
+        Connects a given protocol factory to the given numeric TCP/IP port and open a page."""
```
Comment (hellais) [inline ooni/ui/web/web.py]: Given the fact that this is a standard interface for multiservices I think these two comments are redundant.

**Comment by donald213** [comment]
Thank hellais, you have been a great help. As far as I understand I have to remove every redundant lines. To be honest I do not think there will stay a lot of my work. I save what I can. Fell free to check out my next submission. See you soon.

File: ooni/ui/web/web.py (hunk 7 above)
Comment (donald213) [inline ooni/ui/web/web.py]: What about stopService()?

**Comment by hellais** [comment]
I went ahead and make some changed on top of your branch and merged it into master.

(I also squashed all your commits into 1 and edited the author, in the future you may want to setup you git so that the username is set properly. You can do so with: `git config --global user.name Donald &&  git config --global user.email donald213@users.noreply.github.com`

**Pull request closed without merging**